tests/effects/tdiagnostic_messages.nim

```diff
@@ -2,19 +2,20 @@ discard """
   nimoutFull: true
   action: "reject"
   cmd: "nim r --hint:Conf:off $file"
-  nimout: '''tdiagnostic_messages.nim(35, 6) Error: 'a' can have side effects
-> tdiagnostic_messages.nim(36, 30) Hint: 'a' calls `.sideEffect` 'callWithSideEffects'
->> tdiagnostic_messages.nim(28, 6) Hint: 'callWithSideEffects' called by `.noSideEffect` 'a'
->>> tdiagnostic_messages.nim(30, 34) Hint: 'callWithSideEffects' calls `.sideEffect` 'indirectCallViaVarParam'
->>>> tdiagnostic_messages.nim(24, 6) Hint: 'indirectCallViaVarParam' called by `.noSideEffect` 'callWithSideEffects'
->>>>> tdiagnostic_messages.nim(25, 7) Hint: 'indirectCallViaVarParam' calls routine via hidden pointer indirection
->>> tdiagnostic_messages.nim(31, 33) Hint: 'callWithSideEffects' calls `.sideEffect` 'indirectCallViaPointer'
->>>> tdiagnostic_messages.nim(26, 6) Hint: 'indirectCallViaPointer' called by `.noSideEffect` 'callWithSideEffects'
->>>>> tdiagnostic_messages.nim(27, 32) Hint: 'indirectCallViaPointer' calls routine via pointer indirection
->>> tdiagnostic_messages.nim(32, 10) Hint: 'callWithSideEffects' calls `.sideEffect` 'myEcho'
->>>> tdiagnostic_messages.nim(23, 6) Hint: 'myEcho' called by `.noSideEffect` 'callWithSideEffects'
->>> tdiagnostic_messages.nim(33, 3) Hint: 'callWithSideEffects' accesses global state 'globalVar'
->>>> tdiagnostic_messages.nim(22, 5) Hint: 'globalVar' accessed by `.noSideEffect` 'callWithSideEffects'
+  nimout: '''
+tdiagnostic_messages.nim(36, 6) Error: 'a' can have side effects
+> tdiagnostic_messages.nim(37, 30) Hint: 'a' calls `.sideEffect` 'callWithSideEffects'
+>> tdiagnostic_messages.nim(29, 6) Hint: 'callWithSideEffects' called by `.noSideEffect` 'a'
+>>> tdiagnostic_messages.nim(31, 34) Hint: 'callWithSideEffects' calls `.sideEffect` 'indirectCallViaVarParam'
+>>>> tdiagnostic_messages.nim(25, 6) Hint: 'indirectCallViaVarParam' called by `.noSideEffect` 'a'
+>>>>> tdiagnostic_messages.nim(26, 7) Hint: 'indirectCallViaVarParam' calls routine via hidden pointer indirection
+>>> tdiagnostic_messages.nim(32, 33) Hint: 'callWithSideEffects' calls `.sideEffect` 'indirectCallViaPointer'
+>>>> tdiagnostic_messages.nim(27, 6) Hint: 'indirectCallViaPointer' called by `.noSideEffect` 'a'
+>>>>> tdiagnostic_messages.nim(28, 32) Hint: 'indirectCallViaPointer' calls routine via pointer indirection
+>>> tdiagnostic_messages.nim(33, 10) Hint: 'callWithSideEffects' calls `.sideEffect` 'myEcho'
+>>>> tdiagnostic_messages.nim(24, 6) Hint: 'myEcho' called by `.noSideEffect` 'a'
```

Comment:
I much prefer the previous messages before last commit; the new way is confusing as you're omitting the direct links; in more complex cases (eg with templates involved or deeper stacks), it'll be hard to tell how is X called by Y because it wouldn't show intermediate links.

Reply:
Yeah :)
It had said ```Hint: 'globalVar' accessed by `.noSideEffect` 'callWithSideEffects'```.
`callWithSideEffects` is `.sideEffect`

I just dropped the `.noSideEffect` if that's fine.

Reply:
good now :)